src/zcash/NoteEncryption.cpp

```diff
@@ -133,7 +133,7 @@ std::optional<SaplingEncCiphertext> SaplingNoteEncryption::encrypt_to_recipient(
 
     crypto_aead_chacha20poly1305_ietf_encrypt(
         ciphertext.begin(), NULL,
-        message.begin(), ZC_SAPLING_ENCPLAINTEXT_SIZE,
+        message.begin(), SAPLING_ENCPLAINTEXT_SIZE,
```

Comment:
So, some of the constants moved to NoteEncryption.hpp because of the new reliance on `Memo::SIZE`. In the move, it seemed like a good idea to change them from `#define` to `constexpr`, which can (and should) be namespaced. Namespacing them makes the `ZC_` redundant in favor of `libzcash::`.

I think that set of changes makes sense to do together. But I could make a commit that does the moving and renaming on its own, then apply the memo changes on top.